app/src/scripts/_utils.js

```diff
@@ -81,7 +81,103 @@ export const addComplete = (quests, valleyName) => {
   else {
     return 0;
   }
-}
+};
+
+export const agesData = [
+  {
+    index: 0,
+    name: "Exploration",
+    image: null,
+    requirements: {
+      Coding: 1,
+      Electronics: 1,
+      ThreeD: 1
+    }
+  },
+  {
+    index: 1,
+    name: "Discovery",
+    image: null,
+    requirements: {
+      total: 5
+    }
+  },
+  {
+    index: 2,
+    name: "Making",
+    image: null,
+    requirements: {
+      MakerMount: "all"
+    }
+  },
+  {
+    index: 3,
+    name: "Sharing",
+    image: null,
+    requirements: {
+      Placeholder: true
+    }
+  }
+];
+
+// export const getAge = (quests, agesData) => {
+//   //console.log(Object.keys(agesData[0].requirements))
+//   let currentAge = {}
+//   agesData.forEach( (ageData) => {
+//     //console.log(ageData)
+//     let newArray = Object.keys(ageData.requirements).map( (valleyName) => {
+//       if (valleyName === "total") {
+//         return addComplete(quests)
+//       } else {
+//         if (addComplete(quests, valleyName, null) === addComplete(quests, valleyName) && typeof(ageData.requirements[valleyName]) === "string"){
+//           console.log()
+//           return "all"
+//         } else {
+//           return addComplete(quests, valleyName)
+//         }
+//       }
+//     });
+//     console.log("A not so new array " + Object.values(ageData.requirements))
+//     if ( (newArray >= Object.values(ageData.requirements) && Object.values(ageData.requirements) !== -1) || newArray === Object.values(ageData.requirements)) {
+//       console.log("Passed Age: " + ageData.name)
+//     } else {
+//       console.log("Failed Age: " + ageData.name)
+//     }
+//     console.log("A whole new array " + newArray)
+//   });
+// };
+
+export const getAge = (quests, agesData) => {
+  //console.log(Object.keys(agesData[0].requirements))
+  for (let i = 0; i < agesData.length; i++) {
+    //console.log(ageData)
+    let newArray = Object.keys(agesData[i].requirements).map( (valleyName) => {
+      if (valleyName === "total") {
+        return addComplete(quests)
+      } else {
+        if (addComplete(quests, valleyName, null) === addComplete(quests, valleyName) && typeof(agesData[i].requirements[valleyName]) === "string"){
```

Comment:
I'd check the typeof first for performance reasons:
```suggestion
        if (typeof(agesData[i].requirements[valleyName]) === "string" && addComplete(quests, valleyName, null) === addComplete(quests, valleyName)){
```